### ISSUE TITLE
Think about VM deployment

### ISSUE THREAD

**Issue by copelco** [caktus/rapidsms-salt#8]
Should we use salt-cloud? Just ssh in as a user with sudo privs and bootstrap salt? Let me know if you need a server to play with.

**Comment by dpoirier** [comment]
Instead of vagrant? It would be nice if one approach would work with both, but I don't see vagrant support in salt-cloud (https://github.com/saltstack/salt-cloud/tree/develop/saltcloud/clouds)

**Comment by copelco** [comment]
Yeah, instead of Vagrant, so we could easily to deploy to Linode (or the like). Maybe we could just have a fab command that runs salt bootstrap and rsync's the templates?

**Comment by dpoirier** [comment]
Yeah, something that just assumed an Ubuntu system we could ssh into and sudo, and then set everything else up, would be usable on a lot of things. We wouldn't need something like salt-cloud.  We might want to add salt bootstrapping to Argyle.

**Comment by copelco** [comment]
Another gotcha: we'll need to have a salt_bootstrap plus something like salt_update if we go with the masterless setup, so we can update the salt files and run the highstate command again. Adding new users, updating salt configs, etc.

**Comment by copelco** [comment]
Ran into some issues with this today on another project:
- `admin` group doesn't have NOPASSWD by default on Amazon AMI. Need to add our own modification of /etc/sudoers.
- Amazon AMI's default user is `ubuntu`. If a command fails during the ssh setup, the ubuntu user might lose ssh access. We've gotten around this by adding `vagrant` to the users conf. What should we do here?

**Comment by dpoirier** [comment]
Good notes. I expect when we start targeting other Ubuntu systems than vagrant, we'll keep finding things we need to configure explicitly that happened to be set the way we wanted them in vagrant.
